pkg/vere/io/ames.c

```diff
@@ -2765,6 +2771,60 @@ _ames_recv_cb(uv_udp_t*        wax_u,
   }
 }
 
+static void natpmp_init(uv_timer_t* handle);
+
+static void
+natpmp_cb(uv_poll_t* handle,
+          c3_i        status,
+          c3_i        events)
+{
+  u3_ames* sam_u = handle->data;
+
+  natpmpresp_t response;
+  c3_i err_i = readnatpmpresponseorretry(&sam_u->nat_u.req_u, &response);
+  if ( NATPMP_TRYAGAIN == err_i ) {
+    return;
+  }
+
+  uv_poll_stop(handle);
+
+  if ( 0 != err_i ) {
+    u3l_log("ames: natpmp error %i", err_i);
+    uv_poll_stop(&sam_u->nat_u.pol_u);
+    closenatpmp(&sam_u->nat_u.req_u);
+    return;
+  }
+
+  u3l_log("ames: mapped public port %hu to localport %hu lifetime %u",
+         response.pnu.newportmapping.mappedpublicport,
+         response.pnu.newportmapping.privateport,
+         response.pnu.newportmapping.lifetime);
+
+  closenatpmp(&sam_u->nat_u.req_u);
+  sam_u->nat_u.tim_u.data = sam_u;
+  uv_timer_start(&sam_u->nat_u.tim_u, natpmp_init, 7200000, 0);
+}
+
+static void
+natpmp_init(uv_timer_t *handle)
+{
+  u3_ames* sam_u = handle->data;
+  c3_s por_s = sam_u->pir_u->por_s;
+
+  c3_i err_i = initnatpmp(&sam_u->nat_u.req_u, 0, 0);
+
+  if (err_i != 0) {
+    return;
+  }
+
+  uv_poll_init(u3L, &sam_u->nat_u.pol_u, sam_u->nat_u.req_u.s);
```

Comment:
It looks like `uv_poll_init()` is categorically different from other libuv init functions, and we need to check for errors here.